### ISSUE TITLE
Unable to customise icon and text in Unity sidebar.

### ISSUE THREAD

**Issue by illume** [pygame/pygame#87]
Originally reported by: **Anonymous**

----------------------------------------

When running a PyGame window with Ubuntu + Unity, the applications icon becomes a blurred Python snake, with the caption "Python v2.7".

Using display.set_icon and display_set_caption doesn't resolve this issue.  It seems these functions only affect the window itself, not Unity.

Screenshot of issue: http://imgur.com/AMDWv

----------------------------------------
- Bitbucket: https://bitbucket.org/pygame/pygame/issue/87

**To do to close**

- [x] Test on ubuntu with unity sidebar that unity sidebar icons are set in a different way to the icons in the corner of windows as explained below.

**Comment by MyreMylar** [comment]
I feel like this is a case of needing a .ico file for a frozen python script. There are two different 'icon' concepts here

1. The Icon that appears in the top left corner of a window. this is set by SDL's SetWindowIcon that pygame wraps with display.set_icon().

2. The icon that appears in the file system that you would double click to run a program. 

These are two distinct things on windows for sure, and I expect on all platforms using SDL. I believe that the Unity launcher is looking for an icon of type 2 and since it doesn't find a specific one for the program (i'm not sure a python script can even have a unique icon of the second type without being frozen) it defaults to the default icon for python script files.

Once a script is frozen you should be able to set an icon for it in the normal way that is done on whichever platform.

In short, nothing we can do here, no user to follow up with - recommend close.

**Comment by MyreMylar** [comment]
Before closing it would be good to get someone on Ubuntu with the Unity sidebar to confirm that my suppositions are correct about how to change an icon on the sidebar with a frozen pygame script.

**Comment by MyreMylar** [comment]
**To do to close**

- [ ] Test on ubuntu with unity sidebar that unity sidebar icons are set in a different way to the icons in the corner of windows as above.

**Comment by robertpfeiffer** [comment]
I suppose we close this, given that compiz-based unity and bamfdaemon have been deprecated.

**Comment by robertpfeiffer** [comment]
For posterity: The "fix" would have been to make a binary with PyInstaller and to then ship a .desktop file with your game that refers to the binary and has an icon. Then bamfdaemon would see the process binary and match it to the .desktop file and the window. Writing .desktop files is totally out of scope for PyGame, but Unity desktop *did not want* per-window icons, but per-application icons, so different windows of the same application could be grouped together.
